File: java-playground/src/main/java/org/gonza/javaplayground/game/car/CarName.java
package org.gonza.javaplayground.game.car;

import java.util.Objects;

public class CarName {
    private static final Integer MAX_NAME_LENGTH = 5;

    private final String name;

    public CarName(String name) {
        if (Objects.isNull(name) || name.length() > MAX_NAME_LENGTH) {
            throw new IllegalArgumentException("Name is too long");
        }

        this.name = name;
    }

    public String getValue() {
        return name;
    }
}

File: kotlin-playground/src/main/kotlin/org/gonza/kotlinplayground/domain/car/CarName.kt
package org.gonza.kotlinplayground.domain.car

class CarName(
    val value: String
) {
    init {
        if(isOverFive(value)) {
            throw IllegalArgumentException("name must be lte 5")
        }
    }

    private fun isOverFive(value: String): Boolean = value.length > 5
}